src/middleware.js

```diff
@@ -80,11 +86,26 @@ function domainChanged(domain) {
     const timeSpentOnDomain = ((Date.now() - data['openedTime'] - data['totalInactiveTime']) / 1000);
     const dateString = new Date(Date.now()).toLocaleDateString();
 
-    // set default value to an empty object
+    // add data to the map
     addElement(dateString, lastDomain, timeSpentOnDomain);
 
-    // update the last domain
-    setLastDomain(domain);
+    // check if the current domain is in the whitelist
+    chrome.storage.sync.get(["whitelist"], function (data) {
```

Comment:
this block confuses me a bit. What is the block supposed to be doing?

Reply:
Set 'LastDomain' object in chrome sync.